source/client_stub.c

```diff
@@ -50,25 +55,64 @@ int rtable_disconnect(struct rtable_t *rtable) {
  */
 int rtable_put(struct rtable_t *rtable, struct entry_t *entry) {
 
+    if (!rtable) return -1;
+    MessageT* msg = (MessageT*) calloc(1, sizeof(struct MessageT*));
+    message_t__init(msg);
+    EntryT* ent = (EntryT*) calloc(1, sizeof(struct EntryT*));
+    entry_t__init(ent);
+    ent->key = entry->key;
+    ent->value.data = entry->value->data;
```

Comment:
I hadn't this noticed before, but it's probably better to do a deep copy into `ent->value.data`
```suggestion
    memcpy(&(ent->value.data), entry->value->data, entry->value->datasize);
```

Reply:
changed